voluncheer/profiles/views/volunteers.py

```diff
@@ -20,6 +21,17 @@ def get_context_data(self, **kwargs):
 
     def form_valid(self, form):
         """Saves the new user and logs them in."""
-        user = form.save()
-        login(self.request, user)
-        return redirect("profile")
+        user = form.save(commit=False)
+        user.is_active = True
+        user.save()
+        volunteer_profile = Volunteer.objects.create(
+            user=user,
+            first_name=form.cleaned_data.get("first_name"),
+            last_name=form.cleaned_data.get("last_name"),
+            date_of_birth=form.cleaned_data.get("date_of_birth"),
+        )
+        volunteer_profile.save()
+        user.is_active = False
```

Comment:
Why do you make the user inactive? It's like deleting the user right?

Reply:
https://stackoverflow.com/questions/29742845/django-how-do-i-use-is-active-of-auth-user-table

Reply:
This is for email verification purposes. We want the user to be unable to login unless they have verified their email successfuly

Reply:
Do we have to set it to true, save the user, then set it back to false, and re-save the user? Is that a required step?

Reply:
Yes, this is required because of the way we implemented our User types. Django now has an error: "Assigning unsaved objects to a ForeignKey, GenericForeignKey, and OneToOneField now raises a ValueError.". When we try to save the user with commit=false, it raised an error when we tried to save the user because of the fact that Volunteer/Organizations use that foreign key. Also, if I tried to only create the Volunteer/Organization profile after the activation link, the data from the form would be lost. To solve this, I first set the user to active so that the organization/volunteer profile could be created without issue. Then I save the organization/volunteer profile and then I save the user; finally I set the is_active to false so that the user must activated their account to be able to login.